rust/gitxetcore/src/xetblob/file_operations/cp_operations.rs

```diff
@@ -63,80 +66,104 @@ async fn build_cp_operation_list(
     for src in sources {
         let src_is_directory;
         let src_path: String;
-
-        if src.contains('*') {
-            return Err(GitXetRepoError::InvalidOperation(
-                "Error: Currently, wildcards are not supported.".to_owned(),
-            ));
-        }
+        let dest_dir: String;
+        let dest_path: String;
+        let mut src_size = 0;
+        let mut file_filter_pattern: String = "".to_string();
 
         // Validate that the source path is specified correctly, and set src_is_directory.
-        let Some(src_info) = src_fs.info(src).await? else {
-            return Err(GitXetRepoError::InvalidOperation(
-                "Error: Source {src_path} does not exist".to_owned(),
-            ));
-        };
-
-        if src.ends_with('/') {
-            src_path = src.strip_suffix('/').unwrap_or(src.as_str()).to_owned();
 
-            if !src_info.is_dir_or_branch() {
+        if src.contains('*') {
+            // Handling wildcard cases
+            // The file card matching works by recasting the operation as a directory -> directory copy,
+            // but with a possible filter pattern.
+
+            // validate
+            // we only accept globs of the for blah/blah/blah/[glob]
+            // i.e. the glob is only in the last component
+            // src_root_dir should be blah/blah/blah here
+            let src_root_dir = src_fs.parent(src.as_str());
+            
+            if src_root_dir.is_some_and(|d| d.contains('*'))   {
                 return Err(GitXetRepoError::InvalidOperation(format!(
-                    "Source path {src} does not exist as a directory."
+                    "Invalid glob {src}. Wildcards can only appear in the last position",
                 )));
             }
-
+            src_path = src_root_dir.unwrap_or("").to_string();
             src_is_directory = true;
+            dest_path = dest_base_path.to_owned();
+            dest_dir = dest_path.clone();
+            file_filter_pattern = if !src_fs.file_name(src.as_str()).is_empty() {
```

Comment:
Set 
`file_filter_pattern = Some(Pattern::new(file_filter_pattern.as_str()).map_err(|e| GitXetRepoError::InvalidOperation(format!("Error in glob pattern {src_file_name}: {e:?})))`?;